### PR TITLE
Update docs enough for new dev onboarding

### LINKED ISSUE #49: Port various website manuals/writeups into one source; update them; remove dead links
Clean up the content of each of these gdocs:
- [x] https://docs.google.com/document/d/1I1DyXKnjYqMY2RY3Kwl7D4bzvwbsxeTkF3U73XX6sdc/edit#
- [x] https://docs.google.com/document/d/1PuuKfxVCeaoqOXoas8dkjHuJsovHFPZjSe6MMGVVKsQ/edit#
- [x] https://docs.google.com/document/d/1xEwITlKRpn4Vd8iWC4l86eQcNpOCUOYKM644vBQKL-s/edit#
- [x] https://docs.google.com/document/d/16bxwyXN5ZFv4aPFMoNAWLVbaKS2S1D_Chk2lK9RvFJc/edit

They each have their own instructions about webinfra, both on the technical and logistics sides. **Consolidate them: They should really be consolidated into the repo readmes and the main gdoc**, so that we don't have to hold on to invalid/obsolete instructions, and so that we don't have to poke around odd places. If a google doc still exists, it should be recursively pointed to and from, via the main google doc. This makes sure everything is discoverable and connected
Also, **remove any content that is incorrect, out-of-date-etc**
Feel free to write up more (and revise) as you see fit. This part isn't a big deal though, and probably not worth the time

finally, for each of these docs:
- [x] https://docs.google.com/document/d/1AlunZNJ9xJ8nHBAiGBR3P-0IbkD9jxN39KtuM2v4oYs/edit
- [x] https://docs.google.com/document/d/1wzxFF308kQoPed11UQvRbzyHz9rykQd69IcVms4uwa4/edit
- [x] https://docs.google.com/document/d/16bxwyXN5ZFv4aPFMoNAWLVbaKS2S1D_Chk2lK9RvFJc/edit

**Go through our google drive, and check for files everywhere that touch on webinfra; they might be in multiple folders. Unify to one folder. Then, clean redundant / duped or old things.** 

Critical cuz it's important. But: _Low-priority_ since new devs have been onboarded, and we all know what we're doing for now. However, after IAP, this would be great to handle.

### LINKED ISSUE #50: Port website manual gdoc into repo
https://github.com/battlecode/galaxy/issues/49
^ See that , for the frontend issue. This issue is about doing the same for backend.
IDK how much of it is relevant -- if any -- but we should do it once, quickly, just to not lose anything. I can do it w someone else, in case yall need help judging what's worth preserving

critical because what's maybe lost is unknown, and maybe there's important stuff

### PR DESCRIPTION
**Pull request by n8kim1**
While there's still much more to write...
This PR updates our docs for onboarding. In particular, the goals are:

- Nothing in the docs is actively wrong (untruthful, misleading, old, obsolete, etc)
- The original `frontend` docs are all ported to new places (or just deleted if obsolete)
- Enough of an outline is present so that during onboarding, if someone takes notes, they can easily slot those notes into a fitting place

Note: Some filenames and paths will have to be rewritten once frontend2 becomes frontend; I'll cross that bridge when we get to it.

Starts addressing #49

### CODE REVIEW

**Review by acrantel — COMMENTED** [review]
thanks for doing this! I put down two comments
I think we can say that Github issues and roadmap are both places to find TODOs. Roadmap has some TODOs that are not listed in issues, and some Github issues are not listed in Roadmap. I'm planning on listing issues in the roadmap when we decide to prioritize them.

**Comment by acrantel** [comment]
Also I see this PR is still listed as draft - is it ready to merge?

**Comment by n8kim1** [comment]
@acrantel contents are ready for review! (there's some linting and GH cleanup left, but don't wanna let that hold you back)

File: frontend2/docs/onboard.md [hunk 1]
```diff
@@ -42,43 +58,43 @@ There's a couple other parts of React that we use heavily, that are not directly
...
 You may want to know how conditional rendering works -- depending on what you do, it might come up. Here's a good explanation: https://reactjs.org/docs/conditional-rendering.html
 
-Note that as you navigate our frontend, the URL in your web browser changes. A naive website implementation would have different html files for each page. But we only have one HTML file...what gives?
-This is thanks to Routes and Switches, from `react-router`. You might want to know how they work, since they might come up too. This will help you to figure out what component is being rendered, based on a URL. For explanation, you can _skim_ https://v5.reactrouter.com/web/api/Route and https://v5.reactrouter.com/web/api/Switch. If there are important details, make sure to check the docs.
+TODO notes about react routes, and how our code takes in URLs and determines what to actually return. Link: https://reactrouter.com/en/main/route/route.
 
 ### JQuery
 
-#### Technical Detail
+You actually probably don't need to know how JQuery works, other than how its API is exposed as a dollar sign object.
```
Comment (acrantel) [inline frontend2/docs/onboard.md:65]: we don't use jquery anymore, so this section is not relevant anymore

**Review by acrantel — APPROVED** [review]
one comment to delete the jquery section but otherwise this is good to merge

**Pull request merged**